File: DesignPatternsExercises/java/FlyweightDesignPattern/FlyweightDesignPattern2/Solution/FlightFactory.java
import java.util.HashMap;
import java.util.Map;

public class FlightFactory {
    private static final Map<String, Flight> flightMap = new HashMap<>();

    public Flight getFlight(String flightKey) {
        // Flight flight = __________(2)__________.get(flightKey);
        Flight flight = flightMap.get(flightKey);
        if (flight == null) {
            // flight = __________(3)__________(flightKey);
            flight = new Flight(flightKey);
            flightMap.put(flightKey, flight);
        }
        return flight;
    }
}

File: DesignPatternsExercises/ruby/flyweight_design_pattern/flyweight_design_pattern2/exercise/flight_factory.rb
require_relative 'flight'

class FlightFactory
  def __________(3)__________
    __________(4)__________ = {}
  end

  def flight(flight_company)
    return @flight_map[flight_company] if __________(5)__________.key? flight_company

    flight = __________(6)__________(flight_company)
    @flight_map[flight_company] = flight
    flight
  end
end
